### PR TITLE
test: pytest fixture to notify changes in entry-points

### LINKED ISSUE #152: Report if entrypoints are outdated
When using run_demo/run_local/pytest we should detect if the entrypoints are outdated and suggest to redo `pip install -e`.

### PR DESCRIPTION
**Pull request by natthan-pigoux**
Create a pytest fixture to notify devs when entry-points in the environment and the one in pyproject are not consistent.

Closes #152

### CODE REVIEW

File: diracx-testing/src/diracx/testing/__init__.py [hunk 1]
```diff
@@ -642,3 +646,45 @@ def handle_starttag(self, tag, attrs):
...
+        for key, value in eps.items():
+            current_eps[key] = current_eps.get(key, {}) | value
+    return installed_eps == current_eps
+
+
+@pytest.fixture(scope="session", autouse=True)
+def verify_entry_points(request):
+    if not entry_points_consistency():
+        raise RuntimeError(
+            "Project and installed entry-points are not consistent. You should run `pip install -e .`"
```
Comment (chrisburr) [inline diracx-testing/src/diracx/testing/__init__.py:689]:
```suggestion
            "Project and installed entry-points are not consistent. You should run `pip install -r requirements-dev.txt`"
```

File: diracx-testing/src/diracx/testing/__init__.py [hunk 2]
```diff
@@ -642,3 +646,45 @@ def handle_starttag(self, tag, attrs):
...
+    """Parse entry points from pyproject.toml."""
+    try:
+        with open(toml_file, "rb") as f:
+            pyproject = tomllib.load(f)
+        return pyproject.get("project", {}).get("entry-points", {})
+    except KeyError:
+        return {}
+
+
+DIRACX_TOMLS = ["pyproject.toml"] + glob.glob("diracx-*/pyproject.toml")
```
Comment (chrisburr) [inline diracx-testing/src/diracx/testing/__init__.py:671]:
```suggestion
repo_base = Path(__file__).parent.parent.parent.parent
DIRACX_TOMLS = ["pyproject.toml"] + repo_base.glob("diracx-*/pyproject.toml")
```

**Comment by natthan-pigoux** [comment]
@chrisburr, thanks for the review!

**Comment by chrisburr** [comment]
I think this is ready now but as I pushed to it I'll ping @aldbr or @chaen for a review

**Pull request merged**